frontend/packages/console-shared/src/components/pod/PodRing.tsx

```diff
@@ -73,16 +74,14 @@ const PodRing: React.FC<PodRingProps> = ({
           <PodStatus
             standalone
             data={pods}
-            subTitle={
-              resourceObj.spec.replicas !== resourceObj.status.availableReplicas
-                ? !resourceObj.spec.replicas
-                  ? `pods`
-                  : `scaling to ${resourceObj.spec.replicas}`
-                : resourceObj.spec.replicas > 1 || resourceObj.spec.replicas === 0
-                ? 'pods'
-                : 'pod'
-            }
-            title={resourceObj.status.availableReplicas || '0'}
+            subTitle={podRingLabel(resourceObj, isScalingAllowed).subTitle}
```

Comment:
Above, you can declare your title and subTitle like this:

```ts
const {title, subTitle} = podRingLable(resourceObj, isScalingAllowed);
```

This way we only make one call to the `podRingLabel` function.

Reply:
Also, we should probably be using hooks for a lot of the state we are setting above.

Reply:
Don't worry about adding hooks right now though.